### PR TITLE
Added support for Github Actions

### PR DESCRIPTION
**Pull request by mridang**
## Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] I have assigned the correct milestone or created one if non-existent.
- [ ] I have correctly labeled this pull request.
- [ ] I have linked the corresponding issue in this description.
- [ ] I have updated the corresponding Jira ticket.
- [ ] I have requested a review from at least 2 reviewers
- [ ] I have checked the base branch of this pull request
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

**Comment by supercid** [comment]
Squash on merge

File: Plugin/Catalog/Block/AbstractBlock.php [hunk 1]
```diff
@@ -226,7 +224,7 @@ public function afterGetLastNum($block, $result)
      * @param $result
      * @return int
      */
-    public function afterGetTotalNum($block, $result)
+    public function afterGetTotalNum($block, $result) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Block/AbstractBlock.php:227]: We should define the sniff we want to ignore

File: Plugin/Catalog/Block/AbstractBlock.php [hunk 2]
```diff
@@ -239,7 +237,7 @@ public function afterGetTotalNum($block, $result)
      * @param $result
      * @return int
      */
-    public function afterGetLastPageNum($block, $result)
+    public function afterGetLastPageNum($block, $result) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Block/AbstractBlock.php:240]: We should define the sniff we want to ignore

File: Plugin/Catalog/Block/Pager.php [hunk 3]
```diff
@@ -63,14 +63,13 @@ public function __construct(
...
 
-
     /**
      * @param MagentoPager $pager
      * @param $result
      * @param $param
      * @return bool
      */
-    public function afterIsPageCurrent(MagentoPager $pager, $result, $param)
+    public function afterIsPageCurrent(MagentoPager $pager, $result, $param) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Block/Pager.php:72]: We should define the sniff we want to ignore

File: Plugin/Catalog/Block/Pager.php [hunk 4]
```diff
@@ -122,7 +124,7 @@ public function afterGetFramePages(MagentoPager $pager, $result)
      * @param $result
      * @return bool
      */
-    public function afterIsFirstPage(MagentoPager $pager, $result)
+    public function afterIsFirstPage(MagentoPager $pager, $result) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Block/Pager.php:127]: We should define the sniff we want to ignore

File: Plugin/Catalog/Block/Pager.php [hunk 5]
```diff
@@ -135,7 +137,7 @@ public function afterIsFirstPage(MagentoPager $pager, $result)
      * @param $result
      * @return bool
      */
-    public function afterIsLastPage(MagentoPager $pager, $result)
+    public function afterIsLastPage(MagentoPager $pager, $result) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Block/Pager.php:140]: We should define the sniff we want to ignore

File: Plugin/Catalog/Block/Toolbar.php [hunk 6]
```diff
@@ -132,7 +131,7 @@ public function __construct(
      * @return MagentoToolbar
      * @throws NoSuchEntityException
      */
-    public function afterSetCollection(
+    public function afterSetCollection( // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Block/Toolbar.php:134]: We should define the sniff we want to ignore

File: Plugin/Catalog/Model/Config.php [hunk 7]
```diff
@@ -85,10 +85,11 @@ public function __construct(
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      * @throws NoSuchEntityException
      */
-    public function afterGetAttributeUsedForSortByArray(MagentoConfig $catalogConfig, $options)
+    public function afterGetAttributeUsedForSortByArray(MagentoConfig $catalogConfig, $options) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Model/Config.php:88]: We should define the sniff we want to ignore

File: Plugin/Catalog/Model/Product.php [hunk 8]
```diff
@@ -72,8 +72,8 @@ public function afterGetProductUrl(MagentoProduct $product, $url)
      */
     private function parseExistingQueryParams($url)
     {
-        $parsed = parse_url($url, PHP_URL_QUERY);
-        parse_str($parsed, $result);
+        $parsed = parse_url($url, PHP_URL_QUERY); // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Model/Product.php:75]: We should define the sniff we want to ignore

File: Plugin/Catalog/Model/Product.php [hunk 9]
```diff
@@ -72,8 +72,8 @@ public function afterGetProductUrl(MagentoProduct $product, $url)
...
+        parse_str($parsed, $result); // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Model/Product.php:76]: We should define the sniff we want to ignore

File: Plugin/Catalog/Model/Sortby.php [hunk 10]
```diff
@@ -90,7 +90,7 @@ public function __construct(
      * @throws NoSuchEntityException
      * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
-    public function afterGetAllOptions(MagentoSortby $sortBy, $options)
+    public function afterGetAllOptions(MagentoSortby $sortBy, $options) // phpcs:ignore
```
Comment (hannunosto) [inline Plugin/Catalog/Model/Sortby.php:93]: We should define the sniff we want to ignore

File: Utils/Debug/Product.php [hunk 11]
```diff
@@ -53,7 +53,7 @@ class Product
     /**
      * Product constructor.
      */
-    private function __construct()
+    private function __construct() // phpcs:ignore
```
Comment (hannunosto) [inline Utils/Debug/Product.php:56]: We should define the sniff we want to ignore

File: Utils/Debug/ServerTiming.php [hunk 12]
```diff
@@ -53,7 +53,7 @@ class ServerTiming
     /**
      * ServerTiming constructor.
      */
-    private function __construct()
+    private function __construct() // phpcs:ignore
```
Comment (hannunosto) [inline Utils/Debug/ServerTiming.php:56]: We should define the sniff we want to ignore

File: Block/SegmentMapping.php [hunk 13]
```diff
@@ -86,10 +87,11 @@ public function getNostoAccount()
         try {
             $store = $this->storeManager->getStore();
         } catch (\Exception $e) {
-            $this->logger->exception("Could not get Nosto account ID");
+            $this->logger->error($e->getMessage() . 'Could not get Nosto account ID');
```
Comment (hannunosto) [inline Block/SegmentMapping.php:90]: Why was this changed?

Comment (supercid) [reply Block/SegmentMapping.php:90]: Exception receives an object `\Exception`, error receives a message `String`

Comment (hannunosto) [reply Block/SegmentMapping.php:90]: Ah, my bad!
